### PR TITLE
Added "Muffled Audio" effect

### PR DESCRIPTION
**Pull request by OnlyRealNubs**
Muffles the audio. Makes you sound like you are underwater. Also my first hook, so I probably did something wrong.

### CODE REVIEW

File: ChaosMod/Effects/db/Misc/MiscMuffledAudio.cpp [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#include "stdafx.h"
```
Comment (pongo1231) [inline ChaosMod/Effects/db/Misc/MiscMuffledAudio.cpp:1]:
```suggestion
#include <stdafx.h>

```

File: ChaosMod/Effects/db/Misc/MiscMuffledAudio.cpp [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+{
+	Hooks::SetAudioClearness(0);
+}
+
+static void OnStop()
+{
+	Hooks::ResetAudioClearness();
+}
+
+REGISTER_EFFECT(OnStart, OnStop, nullptr, EffectInfo
```
Comment (pongo1231) [inline ChaosMod/Effects/db/Misc/MiscMuffledAudio.cpp:14]:
```suggestion
// clang-format off
REGISTER_EFFECT(OnStart, OnStop, nullptr, EffectInfo
```

File: ChaosMod/Memory/Hooks/AudioClearnessHook.cpp [hunk 3]
```diff
@@ -0,0 +1,41 @@
+#include "stdafx.h"
```
Comment (pongo1231) [inline ChaosMod/Memory/Hooks/AudioClearnessHook.cpp:1]:
```suggestion
#include <stdafx.h>
```

**Review by pongo1231 — CHANGES_REQUESTED** [review]

**Comment by OnlyRealNubs** [comment]
I don't know why this just closed. But, I have merged the suggested commits and fixed merge conflicts.

File: ChaosMod/Memory/Hooks/AudioClearnessHook.cpp [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+#include "Hook.h"
+
+static bool ms_bEnabledHook = false;
+static int ms_fValue = 0;
+
+__int64(*_OG_rage__audRequestedSettings__SetClearness)(__int64 _this, __int64 pitch);
+__int64 _HK_rage__audRequestedSettings__SetClearness(__int64 _this, __int64 pitch)
+{
+	return _OG_rage__audRequestedSettings__SetClearness(_this, ms_bEnabledHook ? ms_fValue : pitch);
+}
```
Comment (pongo1231) [inline ChaosMod/Memory/Hooks/AudioClearnessHook.cpp:12]:
```suggestion
__int64(*_OG_rage__audRequestedSettings__SetClearness)(__int64 _this, __int64 clearness);
__int64 _HK_rage__audRequestedSettings__SetClearness(__int64 _this, __int64 clearness)
{
	return _OG_rage__audRequestedSettings__SetClearness(_this, ms_bEnabledHook ? ms_fValue : clearness);
}
```
Not sure pitch would fit in this scenario?

Comment (OnlyRealNubs) [reply ChaosMod/Memory/Hooks/AudioClearnessHook.cpp]: sloppy copy paste LOL

**Pull request merged**